pkg/client/client.go

```diff
@@ -0,0 +1,131 @@
+// Package client work with server api and set/get keys to it
+package client
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	api "github.com/go-itools-internship/go-secret/pkg/http"
+)
+
+type client struct {
+	options options
+	url     string // address where client will be work with server
+}
+
+// options client
+type options struct {
+	c *http.Client
+}
+
+var defaultClient = options{
+	c: &http.Client{Timeout: 20 * time.Second},
+}
+
+type Option func(o *options)
+
+// Client provide any http client
+func Client(c http.Client) Option {
```

Comment:
No need to copy `http.Client` if we make use of a pointer.
```suggestion
func Client(c *http.Client) Option {
```